vm.go

```diff
@@ -701,6 +769,82 @@ func (vm *VirtualMachineSpec) String() string {
 	return string(jsonString)
 }
 
+func (vm *VirtualMachineSpec) modifySetting(ctx context.Context, doc *hcsschema.ModifySettingRequest) (err error) {
+	if doc.GuestRequest == nil || vm.gc == nil {
+		return vm.system.Modify(ctx, doc)
+	}
+
+	hostdoc := *doc
+	hostdoc.GuestRequest = nil
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Add {
+		err = vm.system.Modify(ctx, &hostdoc)
+		if err != nil {
+			return fmt.Errorf("adding VM resources: %s", err)
+		}
+		defer func() {
+			if err != nil {
+				hostdoc.RequestType = requesttype.Remove
+				rerr := vm.system.Modify(ctx, &hostdoc)
+				if rerr != nil {
+					log.G(ctx).WithError(err).Error("failed to roll back resource add")
+				}
+			}
+		}()
+	}
+	err = vm.gc.Modify(ctx, doc.GuestRequest)
+	if err != nil {
+		return fmt.Errorf("guest modify: %s", err)
+	}
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Remove {
+		err = vm.system.Modify(ctx, &hostdoc)
+		if err != nil {
+			err = fmt.Errorf("removing VM resources: %s", err)
+			log.G(ctx).WithError(err).Error("failed to remove host resources after successful guest request")
+			return err
+		}
+	}
+	return nil
+}
+
+func (vm *VirtualMachineSpec) listenVsock(port uint32) (net.Listener, error) {
+	return winio.ListenHvsock(&winio.HvsockAddr{
+		VMID:      vm.runtimeId,
+		ServiceID: winio.VsockServiceID(port),
+	})
+}
+
+// acceptAndClose accepts a connection and then closes a listener. If the
+// context becomes done or the VM terminates, the operation will be
+// cancelled (but the listener will still be closed).
+func (vm *VirtualMachineSpec) acceptAndClose(ctx context.Context, l net.Listener) (net.Conn, error) {
+	var conn net.Conn
+	ch := make(chan error)
+	go func() {
+		var err error
+		conn, err = l.Accept()
+		ch <- err
+	}()
+
+	select {
+	case err := <-ch:
+		l.Close()
+		return conn, err
+	case <-ctx.Done():
+	}
+	l.Close()
+	err := <-ch
+	if err == nil {
+		return conn, err
+	}
+
+	// Prefer context error to VM error to accept error in order to return the
```

Comment:
nit: rephrase comment - something like below:
Prefer following return order of error: Context error over VM error...